src/runtime_src/core/edge/user/shim.cpp

```diff
@@ -1419,11 +1452,19 @@ getAieArray()
   return aieArray.get();
 }
 
+zynqaie::Aied*
+shim::
+getAied()
+{
+  return aied.get();
+}
+
 void
 shim::
 registerAieArray()
 {
   aieArray = std::make_unique<zynqaie::Aie>(mCoreDevice);
+  aied = std::make_unique<zynqaie::Aied>(mCoreDevice);
```

Comment:
Ahh, aied does not participate in ownership of the device, the core device is valid as long as aied is alive.  Therefore just pass `mCoreDevice.get()` to the constructor.

Reply:
done